<a href="https://colab.research.google.com/github/GOPIKAV77/Python/blob/main/T-test_and_F-Test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**F**- **Test** **and** **T**-**Test**

**F**-**Test**:An F-test is used to verify whether the variance among different groups is significantly different.

**T**-**Test**: A T-test is used to compare the means of different groups .

**In** **ANOVA**

In Anova, F-test is used to determine the significance of the model and T-test is used to determine the significance of the groups, that is which group differ.

An F-Test checks whether a significant relationship exists between dependent variables and a set of independent variables.

T-Test checks the significance of each of the individual independent variables.

**F-TEST**

*y = Bo + B1 X1 + B2 X2 + ......+ Bn Xn*.

Null Hypothesis:           *Ho: Bo = B1 = B2 = .......= Bn=0*

Alternate Hypothesis:      *Ha: one or more Bi not equal to 0.*

TEST STATISTIC:  

F = MSR/MSE, MSR - mean square due to regression, MSE- Mean square due to error.

REJECTION RULE:

using P-value :  Reject *Ho* if p-value less than or equal to alpha

using critical value: Reject *Ho* if *F* is greater than *Fo*

where *Fo* is based on *F*-distribution with P degrees of freedom in the numerator and n-p-1 degrees of freedom in the denominator.



**T-Test**

*y = Bo + B1 X1 + B2 X2 + .... + Bn Xn.*

Null Hypothesis:     *Ho: Bo= B1= B2=....= Bn=0*

Alternate Hypothesis:      *Ha : one or more Bi not equal to 0, i= 1,2,3,...,n*

**TEST STATISTIC:**

t = bi/Sbi, bi is the coefficient of predictor variable and Sbi is the standard error of coefficient.

**REJECTION RULE:**

using p-value :  Reject *Ho* if p-value is less than or equal to alpha

using critical-value: Reject *Ho* if  *t* is less than *-t_alpha/2*  or if *t* greater than or equal to *t_alpha/2*, where *t_alpha/2* - *t*-distribution with n-p-1 degrees of freedom.

In [5]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
df = pd.read_excel("TRUCKING.xlsx")

# F Test

In [4]:
m = ols("travel_time~x1+n_of_deliveries",data=df).fit()
m.summary()

/usr/local/lib/python3.11/dist-packages/scipy/stats/_axis_nan_policy.py:531: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=10
  res = hypotest_fun_out(*samples, **kwds)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            travel_time   R-squared:                       0.904
Model:                            OLS   Adj. R-squared:                  0.876
Method:                 Least Squares   F-statistic:                     32.88
Date:                Fri, 21 Feb 2025   Prob (F-statistic):           0.000276
Time:                        13:26:46   Log-Likelihood:                -6.8398
No. Observations:                  10   AIC:                             19.68
Df Residuals:                       7   BIC:                             20.59
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
===================================================================================
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept          -0.8687      0.952     -0.913      0.392      -3.119       1.381
x1                  0.0611      0.010      6.182      0.000       0.038       0.085
n_of_deliveries     0.9234      0.221      4.176      0.004       0.401       1.446
==============================================================================
Omnibus:                        0.039   Durbin-Watson:                   2.515
Prob(Omnibus):                  0.981   Jarque-Bera (JB):                0.151
Skew:                           0.074   Prob(JB):                        0.927
Kurtosis:                       2.418   Cond. No.                         435.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

from above code we get, **F-statistic:	32.88**, **Prob (F-statistic):	0.000276**

In [6]:
print(anova_lm(m)) # anova value

                  df     sum_sq    mean_sq          F    PR(>F)
x1               1.0  15.871304  15.871304  48.315660  0.000221
n_of_deliveries  1.0   5.729252   5.729252  17.441075  0.004157
Residual         7.0   2.299443   0.328492        NaN       NaN


# T-Test

In [7]:
r = ols("travel_time~x1+n_of_deliveries",data = df).fit()
r.summary()

/usr/local/lib/python3.11/dist-packages/scipy/stats/_axis_nan_policy.py:531: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=10
  res = hypotest_fun_out(*samples, **kwds)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            travel_time   R-squared:                       0.904
Model:                            OLS   Adj. R-squared:                  0.876
Method:                 Least Squares   F-statistic:                     32.88
Date:                Fri, 21 Feb 2025   Prob (F-statistic):           0.000276
Time:                        13:30:33   Log-Likelihood:                -6.8398
No. Observations:                  10   AIC:                             19.68
Df Residuals:                       7   BIC:                             20.59
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
===================================================================================
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept          -0.8687      0.952     -0.913      0.392      -3.119       1.381
x1                  0.0611      0.010      6.182      0.000       0.038       0.085
n_of_deliveries     0.9234      0.221      4.176      0.004       0.401       1.446
==============================================================================
Omnibus:                        0.039   Durbin-Watson:                   2.515
Prob(Omnibus):                  0.981   Jarque-Bera (JB):                0.151
Skew:                           0.074   Prob(JB):                        0.927
Kurtosis:                       2.418   Cond. No.                         435.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The t-test values are,





---

```


coef	std err	t	P>|t|	[0.025	0.975]
Intercept	-0.8687	0.952	-0.913	0.392	-3.119	1.381
x1	0.0611	0.010	6.182	0.000	0.038	0.085
n_of_deliveries	0.9234	0.221	4.176	0.004	0.401	1.446


In [8]:
print(anova_lm(r))#anova value

                  df     sum_sq    mean_sq          F    PR(>F)
x1               1.0  15.871304  15.871304  48.315660  0.000221
n_of_deliveries  1.0   5.729252   5.729252  17.441075  0.004157
Residual         7.0   2.299443   0.328492        NaN       NaN
